src/test/resources/com/puppycrawl/tools/checkstyle/checks/blocks/rightcurly/InputRightCurlyCaseBlocksInSwitchStatementALoneOrSingleline.java

```diff
@@ -0,0 +1,111 @@
+/*
+RightCurly
+option = ALONE_OR_SINGLELINE
+tokens = LITERAL_CASE
+
+*/
+package com.puppycrawl.tools.checkstyle.checks.blocks.rightcurly;
+
+public class InputRightCurlyCaseBlocksInSwitchStatementALoneOrSingleline {
+    public static void test0() {
+        int mode = 0;
+        switch (mode) {
+            case 1: {
+                int x = 1;
+                break;
+            }
+            case 2: {
+                int x = 0;
+                break;
+            }
+            case 3: {
+                break;
+            }
+        }
+    }
+
+    public static void test() {
+        int mode = 0;
+        switch (mode) {
+            case 1:{
+                int x = 1;
+                break;
+            } default :          // violation '}' at column 13 should be alone on a line'
+                int x = 0;
+        }
+    }
+
+    public static void test1() {
+        int k = 0;
+        switch (k) {
+            case 1:{
+                 int x = 1;
+                 break;
+            } case 2:        // violation '}' at column 13 should be alone on a line'
+                 int x = 2;
+                 break;
+        }
+    }
+
+    public static void test2() {
+         int k = 0;
+         switch (k) {
+            case 1:{
+                 int x = 1;
+                 break;
+            }
+            case 2:
+                 int x = 2;
+                 break;
+         }
+    }
+
+    public static void test3() {
+         int k = 0;
+         switch (k) {
+            case 1:{
+                 int x = 1;
+                 break;
+            }
+            case 2:{
+                    int x = 2;
+                    break;
+            } }      // violation '}' at column 13 should be alone on a line'
+    }
+
+    public static void test4() {
+        int mode = 0;
+        switch (mode) { case 0: {int x = 1;} break; default : int x = 5; }
+        // violation above '}' at column 44 should be alone on a line'
+    }
+
+    public static void test5() {
+        int mode = 0;
+        switch (mode) { case 0: int x = 1; }
+    }
+
+    public static void test6() {
+        int mode = 0;
+        switch (mode) {
+            case 0: {int x = 1; break;}  // ok , single line
+            default : break;
+        }
+    }
+
+    public static void test7() {
+        int mode = 0;
+        switch (mode) { case 0: {int x = 1;}     // ok, single line
+            break; default : break; }
+    }
+
+    public static void test8() {
+        int mode = 0;
+        switch (mode) {
+            case 0: int x = 1;
+            case 1: x = 1; break;
+            case 2: {x = 1; break;} // ok, single line
```

Comment:
I am not seeing this so please add or point to where it is. This should be done for all the options, both regular switches and expression switches.

Cases where there are statements before, after, or both the `{}`s.

Example:
{x = 1;} break;
x = 1; { break; }
Etc...

Some weird cases where `{}` is on lines with expression but not on the same line as the other `{}`.

Example:
{
 x = 1; }
{ x = 1;
}
{ x =
1; }

Reply:
> Cases where there are statements before, after, or both the `{}`s.

done add some case like this but in switch statement only  it won't compile in switch rule all statements after `->` has to be enclosed in one block



> Some weird cases where `{}` is on lines with expression but not on the same line as the other `{}`.

add cases like this also